sdk/storage/Azure.Storage.Files.Shares/tests/FileClientTests.cs

```diff
@@ -2006,33 +2004,33 @@ public async Task UploadAsync_ReadOnlyError()
             string shareName = GetNewShareName();
 
             await using DisposingShare test = await GetTestShareAsync(shareName: shareName);
-            ShareFileClient fileClient = InstrumentClient(
-                test.Share.GetRootDirectoryClient().GetFileClient(GetNewFileName()));
+            ShareFileClient fileClient = InstrumentClient(test.Share.GetRootDirectoryClient().GetFileClient(GetNewFileName()));
 
-            await fileClient.CreateAsync(Constants.KB);
+            await fileClient.CreateAsync(size);
             ShareSasBuilder sasBuilder = new ShareSasBuilder
             {
                 ShareName = shareName,
                 Resource = "f",
                 FilePath = fileClient.Path,
-                ExpiresOn = DateTimeOffset.UtcNow.AddMinutes(1)
+                ExpiresOn = Recording.UtcNow.AddHours(+1)
             };
             sasBuilder.SetPermissions(ShareFileSasPermissions.Read);
             UriBuilder sasUri = new UriBuilder(fileClient.Uri)
             {
-                Query = sasBuilder.ToSasQueryParameters(
-                    new StorageSharedKeyCredential(
+                Query = sasBuilder.ToSasQueryParameters(new StorageSharedKeyCredential(
                         TestConfigDefault.AccountName,
                         TestConfigDefault.AccountKey)).ToString()
             };
 
-            ShareFileClient readOnlyClient = new ShareFileClient(new Uri(sasUri.ToString()));
+            ShareFileClient readOnlyClient = new ShareFileClient(new Uri(sasUri.ToString()), GetOptions());
```

Comment:
Should this be instrumented?

Reply:
`GetOptions()` returns a `InstrumentClientOptions`, so it should be instrumented.

Reply:
@tg-msft has told me before that we also need to write the new Clients in `InstrumentClient()`.